src/collection.js

```diff
@@ -464,9 +464,9 @@ export default class Collection {
       return reject("Extraneous Id; can't create a record having one set.");
     }
     const newRecord = {...record,
-      id:      options.synced ||
-                   options.useRecordId ? record.id :
-                                     this.idSchema.generate(),
+      id: options.synced ||
+                       options.useRecordId ? record.id :
+                       this.idSchema.generate(record),
```

Comment:
I think @leplatrem means a variable representing the boolean condition, plus perhaps one for the entire `id` field if you can't get it to fit on one line.